shunting-yard-replicator/src/test/java/com/expediagroup/shuntingyard/replicator/exec/messaging/MessageReaderAdapterTest.java

```diff
@@ -359,12 +363,13 @@ public void metadataOnlySyncEventForTable() {
   }
 
   @Test
-  public void metadataOnlySyncEventForTableWithNullLocation() {
+  public void insertOverwriteEventForUnpartitionedTable() throws MetaException, NoSuchObjectException, TException {
     Optional<MessageEvent> event = newMessageEvent(apiaryAlterTableEvent);
     when(messageReader.read()).thenReturn(event);
     configureMockedEvent(apiaryAlterTableEvent);
 
-    when(apiaryAlterTableEvent.getTableLocation()).thenReturn(null);
+    when(metaStoreClient.getTable(TEST_DB, TEST_TABLE)).thenReturn(hiveTable);
+    when(hiveTable.getPartitionKeys()).thenReturn(Lists.newArrayList());
```

Comment:
Just missing the case where the list of partition keys returned is null, but perhaps thats not even possible? probably fine to leave this out.

Reply:
I think it can that's why I suggested `sourceTable.getPartitionKeysSize() > 0;` as that does the null check in the table class.

Reply:
It doesn't return `null`. It returns an empty list but I will still add a test case.